lib/json_api_resource/resource.rb

```diff
@@ -81,11 +81,22 @@ def catch_errors
     def self.method_missing(method, *args, &block)
       if match = method.to_s.match(/^(.*)=$/)
         self.client_klass.send(match[1], args.first)
+      
       elsif self.client_klass.respond_to?(method.to_sym)
-        self.client_klass.send(method, *args)
+        results = self.client_klass.send(method, *args)
+
+        if results.is_a? JsonApiClient::ResultSet
+          results.map! do |result|
+            self.new(:client => result)
+          end
```

Comment:
1. will this affect `def find` and `def where`?
2. should this be a method to break it out and be used elsewhere?